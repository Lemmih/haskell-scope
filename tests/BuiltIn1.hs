module BuiltIn1 where

x :: ()
x = ()

y :: [()]
y = []

z :: [()] -> ()
z [x] = x
