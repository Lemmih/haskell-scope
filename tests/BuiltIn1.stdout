Scope errors:

Definitions:
  Definition 1:
    [4mx[0m = ()
  Definition 2:
    [4my[0m = []
  Definition 3:
    [4mz [x] = x[0m
  Definition 4:
    z [[4mx[0m] = x

Use sites:
  Definition used: 1
    [4mx[0m :: ()
  Definition used: 2
    [4my[0m :: [()]
  Definition used: 3
    [4mz[0m :: [()] -> ()
  Definition used: 3
    [4mz[0m [x] = x
  Definition used: 4
    z [x] = [4mx[0m
